### PR TITLE
fix a bug when the input of max is not contiguous

### PR DESCRIPTION
**Pull request by haowen-han**
### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/ops/max.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def max_kernel(
 
 def max(inp):
     logging.debug("GEMS MAX")
+    if not inp.is_contiguous():
+        inp = inp.contiguous()
```
Comment (zhzhcookie) [inline src/flag_gems/ops/max.py]: Thanks very much.
There is a contiguous check in `contiguous()`, so just `inp = inp.contiguous()` is enough.

Comment (haowen-han) [reply src/flag_gems/ops/max.py]: done

File: tests/test_general_reduction_ops.py [hunk 2]
```diff
@@ -123,6 +123,20 @@ def test_accuracy_max_without_dim(shape, dtype):
     gems_assert_equal(res_out, ref_out)
 
 
+@pytest.mark.max
+@pytest.mark.parametrize("shape", REDUCTION_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_accuracy_max_without_dim_uncontiguous(shape, dtype):
+    inp = torch.randn(shape, dtype=dtype, device="cuda")[::2,::2]
```
Comment (zhzhcookie) [inline tests/test_general_reduction_ops.py]: code-format-check / pre-commit failed here.
Please change it to `[::2, ::2]`. Thanks.

Comment (haowen-han) [reply tests/test_general_reduction_ops.py]: done

**Review by iclementine — APPROVED** [review]

**Pull request merged**